### PR TITLE
feat: Add helmfile tool

### PR DESCRIPTION
**Pull request by nikolaik**
This tries to add the [helmfile](https://github.com/helmfile/helmfile) tool to allow supporting lockfile updates in https://github.com/renovatebot/renovate/pull/19865

I could not manage to run the dockerfile tests for helmfile locally, but bats tests ran fine.

Should I remove the helmfile tag in the initial PR?

Let me know if anything is wierd, and I'll update.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -136,6 +136,7 @@ jobs:
             java,
             jb,
             helm,
+            helmfile,
```
Comment (viceice) [inline .github/workflows/build.yml:139]: revert and add to latest test image instead

File: test/helmfile/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,75 @@
+ARG IMAGE=containerbase/buildpack
```
Comment (viceice) [inline test/helmfile/Dockerfile:1]: just add a simple test here, as you've bats tests, which cover the relevant parts.

https://github.com/containerbase/base/blob/88af9742be55f24b1c8fd6a6cc14f6bbdf7fe7f8/test/latest/Dockerfile#L163-L189

File: src/usr/local/buildpack/tools/v2/helmfile.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+function check_tool_requirements () {
+  check_semver "$TOOL_VERSION" "all"
+}
+
```
Comment (viceice) [inline src/usr/local/buildpack/tools/v2/helmfile.sh:6]:
```suggestion
```

done by defaults

File: src/usr/local/buildpack/tools/v2/helmfile.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+function install_tool () {
+  local versioned_tool_path
+  local file
+  local arch=linux_amd64
+  local helmfile_file="${TOOL_NAME}_${TOOL_VERSION#v}_${arch}.tar.gz"
```
Comment (viceice) [inline src/usr/local/buildpack/tools/v2/helmfile.sh:11]:
```suggestion
  local helmfile_file="${TOOL_NAME}_${TOOL_VERSION}_${arch}.tar.gz"
```
`v` is already trimmed

File: src/usr/local/buildpack/tools/v2/helmfile.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  versioned_tool_path=$(create_versioned_tool_path)
+
+  create_folder "${versioned_tool_path}/bin"
+  file=$(get_from_url "https://github.com/${TOOL_NAME}/${TOOL_NAME}/releases/download/v${TOOL_VERSION}/${helmfile_file}")
+  bsdtar -C "${versioned_tool_path}/bin" -xf "${file}" "${TOOL_NAME}"
+}
+
+function link_tool () {
+  local versioned_tool_path
+
```
Comment (viceice) [inline src/usr/local/buildpack/tools/v2/helmfile.sh:22]:
```suggestion
```
unused 🙃

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by nikolaik** [comment]
Thanks for the review, made a few improvements based on your feedback. Let me know if anything else needs updating

File: test/helmfile/Dockerfile (hunk 2 above)
Comment (viceice) [reply test/helmfile/Dockerfile:1]: remove this file

File: test/latest/Dockerfile [hunk 6]
```diff
@@ -227,6 +227,31 @@ USER 1000:1000
 
 RUN vendir --version | grep "${VENDIR_VERSION}"
 
+#------------------------------------------------------------------
+# testk:  helmfile
+#------------------------------------------------------------------
```
Comment (viceice) [inline test/latest/Dockerfile:232]: please join to `testj`

File: test/latest/Dockerfile [hunk 7]
```diff
@@ -227,6 +227,31 @@ USER 1000:1000
...
+
+USER 1000
+
+RUN helmfile version | grep "${HELMFILE_VERSION}"
+
+# even more restricted
+USER 1000:1000
+
+RUN helmfile version | grep "${HELMFILE_VERSION}"
+
```
Comment (viceice) [inline test/latest/Dockerfile:254]: we don't need this, only makes test slow

**Review by viceice — APPROVED** [review]

**Pull request merged**